modules/google-maps/src/utils.js

```diff
@@ -63,14 +63,17 @@ function createDeckCanvas(overlay) {
  * @param deck (Deck) - a previously created instances
  */
 export function destroyDeckInstance(deck) {
-  const {_googleMap: map, _eventListeners: eventListeners} = deck.props.userData;
+  const {_eventListeners: eventListeners} = deck.props.userData;
 
   // Unregister event listeners
   for (const eventType in eventListeners) {
-    map.removeListener(eventListeners[eventType]);
+    eventListeners[eventType].remove();
   }
 
   deck.finalize();
+
+  // Remove canvas
+  deck.canvas.parentNode.removeChild(deck.canvas);
```

Comment:
Isn't `deck.finalize()` supposed to do this?

Reply:
Only if the canvas is created by the Deck instance.